### PR TITLE
[dash] Adding a page on Flutter's compile modes.

### PR DESCRIPTION
**Pull request by Sfshaza**
@Hixie, can you review? (Or delegate?)

Staged: https://sz-flutter.firebaseapp.com/testing/building-apps

### CODE REVIEW

File: src/testing/building-apps.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+## Debug
+
+In _debug mode_, the app is set up for debugging on the physical
+device, emulator, or simulator. Debug mode means that:
+
+* [Assertions](https://www.dartlang.org/guides/language/language-tour#assert)
+   are enabled.
+* [Observatory](https://dart-lang.github.io/observatory/) is enabled,
+   allowing you to profile timeline data.
```
Comment (Hixie) [inline src/testing/building-apps.md]: i'd say "allowing you to use the dart debugger" or some such. profiling timeline data in debug mode is a terrible idea. that's what profile mode is for.

File: src/testing/building-apps.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+* Compilation is optimized for fast startup, fast execution, and small
+  package sizes.
+* Service extensions are disabled.
+
+The command `flutter run --release` compiles to release mode.
+Your IDE also supports these modes.  Android Studio, for example,
+provides a **Run > Run...** menu option, as well as a green bug
+icon overlayed with a small triangle on the project page.
+(The menu item shows a pic of the corresponding icon.)
+
```
Comment (Hixie) [inline src/testing/building-apps.md]: maybe mention that `flutter build` is how you build an app for release? or point to the docs on releasing?

File: src/testing/building-apps.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+## Profile
+
+In _profile mode_, some debugging ability is maintained&mdash;enough
+to profile your app's performance. Profile mode is disabled on
+the emulator and simulator, because their behavior is not representative
+of real performance. Profile mode is similar to release mode, with
+the following differences:
+
+* Some service extensions, such as the one that enables the performance
+  overlay, is enabled.
```
Comment (Hixie) [inline src/testing/building-apps.md]: is->are

File: src/testing/building-apps.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+* Tracing is enabled, and Observatory can, most likely, connect to
```
Comment (Hixie) [inline src/testing/building-apps.md]: most likely? :-)

**Comment by Hixie** [comment]
LGTM

File: src/testing/building-apps.md (hunk 4 above)
Comment (Sfshaza) [reply src/testing/building-apps.md]: LOL. OK, deleted.  (Though that's what it says in the wiki.)

File: src/testing/building-apps.md (hunk 1 above)
Comment (Sfshaza) [reply src/testing/building-apps.md]: Good point

File: src/testing/building-apps.md (hunk 3 above)
Comment (Sfshaza) [reply src/testing/building-apps.md]: Oy

**Pull request merged**